### ISSUE TITLE
Revisit bot support in the ES repo

### ISSUE THREAD

**Issue by pugnascotia** [elastic/elasticsearch#62481]
We rely on various GitHub bots to automate actions in the ES repo. Where possible we should move the code and configuration for these actions into the ES repo itself, so that the ES team can take greater control of the automation that the team relies upon.

This issue covers:

   * Enforcing that all PRs that have at least one change type label (starts with `>`).
   * Enforcing that all PRs have at least one area label (starts with `:`)
   * ~~Adding `Team:*` labels in response to the area labels (starting with `:`)~~
   * Adding a label to indicate PRs from the community
   * Any PR labelled `release highlight` must be automatically labelled `>docs` as well.
   * All PRs must have at least one version. This rule also includes backport PRs.

Some of the above won't apply to issues awaiting triage, or backport PRs. The bot needs to be able to delete any messages that it leaves on PRs when labelling issues are addressed.

In particular, we need support the effort to rework changelog generation in #67335. Whenever a PR is created or updated:

   * If the PR's labels require it to have a changelog file, create one or update it from the PR's labels
   * If the PR's labels prohibit it from having a changelog file but one exists, delete it

This issue excludes:

   * Triggering CI actions
   * Triggering CLA checks
   * Adding `Team:*` labels in response to the area labels (starting with `:`) - this is currently handled by `elasticmachine`.

There is prior art for implementing such rules using GitHub Actions, so we can draw on them for inspiration.

**Comment by mark-vieira** [comment]
@pugnascotia It's not clear what all we want to accomplish here. I'd like us to avoid creating issues with really large and ambiguous scope like this as in practice they just live around forever and never get resolved. As I see it there are two main things here:

1. Making it easier to modify the existing automation that exists here. That is, the github labeler bot that is owned by infra.
2. Extending the capabilities of that bot to _enforce_ certain labeling conventions which is currently not done. 

I'm not convinced that reimplementing (1) is really all that important. That bot is pretty universally used across the stack and I think it accomplishes it's job well enough.

Item (2), as I understand it, is currently a capability we _lack_, and one major motivation for this is that lack of consistent labelling is one thing that makes release notes generation a big pain.

Is my account here accurate?

**Comment by pugnascotia** [comment]
I talked to Infra about using the extending the exist bot's capabilities, but essentially they don't have the bandwidth to pick that up. I talked with @rjernst and we agreed that in the longer term, it would be more sustainable and flexible if the ES team were able to take over our bot usage.

Of course it's isn't essential that the ES team have complete control over the bot code, but we are hurting from a lack of features. For example, the release documentation process would definitely be easier if we enforced better labelling. There's still the issue over Infra (not) having time though.
